server/src/MessageHandler.ts

```diff
@@ -5,18 +5,52 @@ import {
   Range,
   Location,
   DeclarationParams,
+  CompletionParams,
+  CompletionList,
+  CompletionItem,
+  Position,
 } from 'vscode-languageserver'
 import * as util from './util'
 import { fullDocumentRange } from './provider'
-import { getDMMF } from '@prisma/sdk'
-import { TextDocument, Position } from 'vscode-languageserver-textdocument'
+import { TextDocument } from 'vscode-languageserver-textdocument'
 import format from './format'
+import {
+  getSuggestionsForAttributes,
+  getSuggestionsForTypes,
+  getSuggestionForBlockTypes,
+  getSuggestionForFirstInsideBlock,
+  getSuggestionForSupportedFields,
+  getSuggestionsForInsideAttributes,
+} from './completions'
 
-function getWordAtPosition(document: TextDocument, position: Position): string {
-  const currentLine = document.getText({
-    start: { line: position.line, character: 0 },
-    end: { line: position.line, character: 9999 },
+export function getCurrentLine(document: TextDocument, line: number): string {
+  return document.getText({
+    start: { line: line, character: 0 },
+    end: { line: line, character: 9999 },
   })
+}
+
+function isFirstInsideBlock(
+  position: Position,
+  document: TextDocument,
+): boolean {
+  const currentLine = getCurrentLine(document, position.line)
+  if (currentLine.trim().length === 0) {
+    return true
+  }
+
+  const stringTillPosition = currentLine.substring(0, position.character).trim()
+  const firstWordInLine = stringTillPosition.replace(/ .*/, '')
```

Comment:
What's the intention behind this regexp? Are you trying to trim after from the space onward?

Reply:
yes, that's a nicer way for sure.